File: leetcode/swordToOffer/003/code.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
    public:
        vector<int> countBits(int n) 
        {   
            vector<int> result = vector<int>(n+1,0);
            for(int i = 1; i<= n ; i = i+1)
            {
                result[i] = result[i>>1] + (i%2); 
            }
            return result;
        }
};

int main()
{
    Solution solution ;
    vector<int> result = solution.countBits(5);
    cout<<result[1];
}

File: leetcode/swordToOffer/024/code.py
# Definition for singly-linked list.
class ListNode(object):
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next
class Solution(object):
    def reverseList(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """

        if(head.next == None or head == None):
            return head
        result = self.reverseList(head.next)
        p = head
        while(p.next != None):
            p = p.next
        p.next = head
        head.next = None
        return result

node1 = ListNode(1)
node2 = ListNode(2)
node3 = ListNode(3)
node4 = ListNode(4)

node1.next = node2
node2.next = node3
node3.next = node4
node4.next = None

b = Solution()
print(b.reverseList(node1).val)